_includes/live-demos/custom-shortcut-2/index.js

```diff
@@ -0,0 +1,31 @@
+tinymce.init({
+  selector: '#custom-shortcut-2',
+  plugins: 'autolink lists link',
+  toolbar: 'undo redo | bold italic link bullist | insertUsername',
+  setup: function (editor) {
+    editor.addShortcut('meta+alt+U', 'Insert username', function () {
+      insertUsername();
+    });
+
+    editor.ui.registry.addMenuButton('insertUsername', {
+      icon: 'plus',
+      fetch: function (callback) {
+        var items = [
+          {
+            type: 'menuitem',
+            text: 'Insert username',
+            shortcut: 'meta+alt+U',
+            onAction: function () {
+              insertUsername();
+            }
+          }
+        ];
+        callback(items);
+      }
+    });
+  }
+});
+
+function insertUsername() {
```

Comment:
We don't normally suggest doing this, so this should probably be moved inside the setup block. eg:

```
setup: function (editor) {
  var insertUsername = function () {
    editor.insertContent(`@username`);
  }
}
```

Reply:
@lnewson You mean using `TinyMCE.activeEditor`? @spocke had a simiar remark for another thing, but our docs is littered with that string. 188 hits over 53 files....

Reply:
Yeah, using `tinymce.activeEditor` can lead to issues in edge cases.